### PR TITLE
Feat up mimetype

### PR DESCRIPTION
**Pull request by wildintellect**
Adds support for additional STAC media types, (#50), some aliases for existing ones, and proposed parquet type. Requires the addition of `intake_geopandas` and `intake_parquet`

Unresolved media types:
* 'application/x-hdf5': '',
* 'application/x-hdf': '',
* 'application/json': '?',
* 'application/geopackage+sqlite3': 'geopandas',

intake_geopandas does support spatialite, so an enhancement there should cover geopackage.

### CODE REVIEW

**Comment by wildintellect** [comment]
TODO: 
* Add test stac records for these mimetypes to the pytest example data.
* Add new dependencies to 3.6 and 3.7-upstream

File: intake_stac/catalog.py [hunk 1]
```diff
@@ -331,15 +331,25 @@ def __init__(self, key, item, stacked=False):
     def _get_driver(self, entry):
         drivers = {
             'application/netcdf': 'netcdf',
+            'application/x-netcdf': 'netcdf',
+            'application/parquet': 'parquet',
+            'application/x-parquet': 'parquet',
+            # 'application/x-hdf5': '',
+            # 'application/x-hdf': '',
```
Comment (jhamman) [inline intake_stac/catalog.py:338]: There is not a specific driver for hdf at this point so we could either leave this off or try the netcdf driver, which will work on a subset of hdf5 datasets.

File: intake_stac/catalog.py [hunk 2]
```diff
@@ -331,15 +331,25 @@ def __init__(self, key, item, stacked=False):
...
+            'image/tiff; application=geotiff': 'rasterio',
+            'image/tiff; application=geotiff; profile=cloud-optimized': 'rasterio',  # noqa: E501
             'image/png': 'xarray_image',
             'image/jpg': 'xarray_image',
             'image/jpeg': 'xarray_image',
             'text/xml': 'textfiles',
             'text/plain': 'textfiles',
             'text/html': 'textfiles',
+            'application/geo+json': 'geopandas',
+            # 'application/json': '?',
```
Comment (jhamman) [inline intake_stac/catalog.py:351]: probably use `textfiles`

File: intake_stac/catalog.py [hunk 3]
```diff
@@ -331,15 +331,25 @@ def __init__(self, key, item, stacked=False):
...
+            # 'application/geopackage+sqlite3': 'geopandas',
```
Comment (jhamman) [inline intake_stac/catalog.py:352]: I don't think intake has a driver for this yet. I suppose it would be geopandas once available though.

Comment (wildintellect) [reply intake_stac/catalog.py:351]: Should we leave it commented out, or just not include for now?

File: intake_stac/catalog.py (hunk 1 above)
Comment (wildintellect) [reply intake_stac/catalog.py]: Lets try maybe it will work for some.

File: intake_stac/catalog.py (hunk 3 above)
Comment (wildintellect) [reply intake_stac/catalog.py:351]: Right so the question is do we leave the stubs or just note it somewhere else like in tickets?

Comment (scottyhq) [reply intake_stac/catalog.py:351]: i'm in favor of leaving the comments in and making a not in an open issue

File: intake_stac/catalog.py [hunk 4]
```diff
@@ -352,7 +362,7 @@ def _get_driver(self, entry):
 
     def _get_args(self, entry, driver, stacked=False):
         args = entry if stacked else {'urlpath': entry.get('href')}
-        if driver in ['netcdf', 'rasterio', 'xarray_image']:
+        if driver in ['netcdf', 'rasterio', 'xarray_image', 'geopandas', 'parquet']:
```
Comment (wildintellect) [inline intake_stac/catalog.py:365]: I realize I'm not actually sure what this line does and which drivers should be included, perhaps only the ones that can read in chunks? so maybe not geopandas, not sure about parquet.

Comment (jhamman) [reply intake_stac/catalog.py:365]: yeah, I think we can leave this line as it was. Intake/fastparquet should provide reasonable defaults for parquet and there is not a chunked geopandas reader (at this time).

File: intake_stac/catalog.py [hunk 5]
```diff
@@ -331,15 +331,25 @@ def __init__(self, key, item, stacked=False):
     def _get_driver(self, entry):
         drivers = {
             'application/netcdf': 'netcdf',
+            'application/x-netcdf': 'netcdf',
+            'application/parquet': 'parquet',
+            'application/x-parquet': 'parquet',
+            'application/x-hdf5': 'netcdf',
+            # 'application/x-hdf': '',
             'image/vnd.stac.geotiff': 'rasterio',
             'image/vnd.stac.geotiff; cloud-optimized=true': 'rasterio',
```
Comment (scottyhq) [inline intake_stac/catalog.py:340]: should these be removed in favor of the new tiff labels below? or are they still valid?

Comment (wildintellect) [reply intake_stac/catalog.py:340]: I chose to not remove and previously used tags that might still exist in STAC catalogs, they could be marked for future deprecation. The new COG tag isn't official yet either, but community adopted for now.

**Pull request merged**